### ISSUE TITLE
"Actions" menu does not open - server meets all requirements

### ISSUE THREAD

**Issue by feschoppe** [imedias/glados#49]
I don’t know what I’m doing wrong, but in my installation of glados the "Actions" menu is empty. After logging in, I can’t do anything because the menu is empty. The server meets all requirements and I used the deb-packages to install glados on debian 9.5. It is installed in a virtual machine. What am I doing wrong?

**Comment by chaoos** [comment]
Hi feschoppe

Do you mean that the whole menu bar is empty? So there is no menu item to click on?
Also, can you give me some extra information, about the installation? Attach the output of the following commands please:
* `/usr/share/glados/yii migrate/history all`
* `grep -i glados /var/log/dpkg.log`
* `dpkg --list | grep glados`
* `uname -a`
* `php --version`
* `apachectl -M`
* The version of glados
* The user you have logged into glados
* A screenshot of http://<your_ip>/glados/requirements.php in a browser

Greetings chaoos

**Comment by feschoppe** [comment]
Hi chaoos,

thanks a lot for your fast response! We really would appreciate to use the glados exam server at our school.

Only the "Actions" sub menu is empty. To clarify what I mean, I attached a screen shot.

This is the output of sudo /usr/share/glados/yii migrate/history all:

```
Yii Migration Tool (based on Yii v2.0.15.1)

Total 15 migrations have been applied before:
	(2018-09-16 19:05:19) m180724_125845_v1_0_4
	(2018-09-16 19:05:19) m180717_133953_issue21
	(2018-09-16 19:05:19) m180717_064440_issue37
	(2018-09-16 19:05:19) m180606_090312_bug17
	(2018-09-16 19:05:19) m180122_104319_v1_0_3
	(2018-09-16 19:05:19) m170906_143652_libreoffice
	(2018-09-16 19:05:19) m170831_125616_screenshots
	(2018-09-16 19:05:19) m170720_070429_download
	(2018-09-16 19:05:19) m170707_092346_results
	(2018-09-16 19:05:19) m170626_085917_load
	(2018-09-16 19:05:19) m170216_090248_config
	(2018-09-16 19:05:18) m160624_071957_eventstream
	(2018-09-16 19:05:18) m160623_095016_initial
	(2018-09-16 19:05:17) m170907_052038_rbac_add_index_on_auth_assignment_user_id
	(2018-09-16 19:05:17) m140506_102106_rbac_init
```

Without root privileges the command fails with:

`Exception 'yii\db\Exception' with message 'SQLSTATE[HY000] [1045] Access denied for user ''@'localhost' (using password: NO)'`

grep -i glados /var/log/dpkg.log:

```
2018-09-15 23:22:47 install glados:all <keine> 1.0.4-deb9
2018-09-15 23:22:47 status half-installed glados:all 1.0.4-deb9
2018-09-15 23:22:47 status unpacked glados:all 1.0.4-deb9
2018-09-15 23:22:47 status unpacked glados:all 1.0.4-deb9
2018-09-15 23:23:26 status unpacked glados:all 1.0.4-deb9
2018-09-15 23:23:26 remove glados:all 1.0.4-deb9 <keine>
2018-09-15 23:23:26 status half-installed glados:all 1.0.4-deb9
2018-09-15 23:24:09 status half-installed glados:all 1.0.4-deb9
2018-09-15 23:24:09 remove glados:all 1.0.4-deb9 <keine>
2018-09-15 23:24:09 status half-installed glados:all 1.0.4-deb9
2018-09-15 23:24:47 install yii2-glados:all <keine> 2.0.15.1-1
2018-09-15 23:24:47 status half-installed yii2-glados:all 2.0.15.1-1
2018-09-15 23:24:54 status unpacked yii2-glados:all 2.0.15.1-1
2018-09-15 23:24:54 status unpacked yii2-glados:all 2.0.15.1-1
2018-09-15 23:24:54 configure yii2-glados:all 2.0.15.1-1 2.0.15.1-1
2018-09-15 23:24:54 status unpacked yii2-glados:all 2.0.15.1-1
2018-09-15 23:24:54 status half-configured yii2-glados:all 2.0.15.1-1
2018-09-15 23:24:54 status installed yii2-glados:all 2.0.15.1-1
2018-09-15 23:25:06 status half-installed glados:all 1.0.4-deb9
2018-09-15 23:25:06 remove glados:all 1.0.4-deb9 <keine>
2018-09-15 23:25:06 status half-installed glados:all 1.0.4-deb9
2018-09-15 23:31:44 status half-installed glados:all 1.0.4-deb9
2018-09-15 23:31:44 remove glados:all 1.0.4-deb9 <keine>
2018-09-15 23:31:44 status half-installed glados:all 1.0.4-deb9
2018-09-15 23:32:52 status half-installed glados:all 1.0.4-deb9
2018-09-15 23:32:52 remove glados:all 1.0.4-deb9 <keine>
2018-09-15 23:32:52 status half-installed glados:all 1.0.4-deb9
2018-09-16 19:01:50 upgrade glados:all 1.0.4-deb9 1.0.4-deb9
2018-09-16 19:01:50 status half-installed glados:all 1.0.4-deb9
2018-09-16 19:01:51 status half-installed glados:all 1.0.4-deb9
2018-09-16 19:01:51 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:01:51 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:04:48 configure glados:all 1.0.4-deb9 <keine>
2018-09-16 19:04:48 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:04:48 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:04:48 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:04:48 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:04:48 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:04:48 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:04:48 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:04:48 status half-configured glados:all 1.0.4-deb9
2018-09-16 19:05:49 configure glados:all 1.0.4-deb9 <keine>
2018-09-16 19:05:49 status half-configured glados:all 1.0.4-deb9
2018-09-16 19:09:27 configure glados:all 1.0.4-deb9 <keine>
2018-09-16 19:09:27 status half-configured glados:all 1.0.4-deb9
2018-09-16 19:09:30 status installed glados:all 1.0.4-deb9
2018-09-16 19:53:41 upgrade yii2-glados:all 2.0.15.1-1 2.0.15.1-1
2018-09-16 19:53:41 status half-configured yii2-glados:all 2.0.15.1-1
2018-09-16 19:53:41 status unpacked yii2-glados:all 2.0.15.1-1
2018-09-16 19:53:41 status half-installed yii2-glados:all 2.0.15.1-1
2018-09-16 19:53:50 status half-installed yii2-glados:all 2.0.15.1-1
2018-09-16 19:53:50 status unpacked yii2-glados:all 2.0.15.1-1
2018-09-16 19:53:50 status unpacked yii2-glados:all 2.0.15.1-1
2018-09-16 19:53:50 configure yii2-glados:all 2.0.15.1-1 2.0.15.1-1
2018-09-16 19:53:50 status unpacked yii2-glados:all 2.0.15.1-1
2018-09-16 19:53:50 status half-configured yii2-glados:all 2.0.15.1-1
2018-09-16 19:53:50 status installed yii2-glados:all 2.0.15.1-1
2018-09-16 19:54:34 upgrade glados:all 1.0.4-deb9 1.0.4-deb9
2018-09-16 19:54:34 status half-configured glados:all 1.0.4-deb9
2018-09-16 19:54:34 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:54:34 status half-installed glados:all 1.0.4-deb9
2018-09-16 19:54:34 status half-installed glados:all 1.0.4-deb9
2018-09-16 19:54:34 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:54:34 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:54:34 configure glados:all 1.0.4-deb9 1.0.4-deb9
2018-09-16 19:54:34 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:54:34 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:54:34 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:54:34 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:54:34 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:54:34 status unpacked glados:all 1.0.4-deb9
2018-09-16 19:54:34 status half-configured glados:all 1.0.4-deb9
2018-09-16 19:54:37 status installed glados:all 1.0.4-deb9
2018-09-16 20:41:55 upgrade yii2-glados:all 2.0.15.1-1 2.0.15.1-1
2018-09-16 20:41:55 status half-configured yii2-glados:all 2.0.15.1-1
2018-09-16 20:41:55 status unpacked yii2-glados:all 2.0.15.1-1
2018-09-16 20:41:55 status half-installed yii2-glados:all 2.0.15.1-1
2018-09-16 20:42:04 status half-installed yii2-glados:all 2.0.15.1-1
2018-09-16 20:42:04 status unpacked yii2-glados:all 2.0.15.1-1
2018-09-16 20:42:04 status unpacked yii2-glados:all 2.0.15.1-1
2018-09-16 20:42:04 configure yii2-glados:all 2.0.15.1-1 2.0.15.1-1
2018-09-16 20:42:04 status unpacked yii2-glados:all 2.0.15.1-1
2018-09-16 20:42:04 status half-configured yii2-glados:all 2.0.15.1-1
2018-09-16 20:42:04 status installed yii2-glados:all 2.0.15.1-1
```

dpkg --list | grep glados:

```
ii  glados                               1.0.4-deb9                        all          The exam server webinterface
ii  yii2-glados                          2.0.15.1-1                        all          yii2 and plugins needed for glados
```

uname -a:

`Linux glados 4.9.0-8-amd64 #1 SMP Debian 4.9.110-3+deb9u4 (2018-08-21) x86_64 GNU/Linux`

php --version:

```
PHP 7.0.30-0+deb9u1 (cli) (built: Jun 14 2018 13:50:25) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.30-0+deb9u1, Copyright (c) 1999-2017, by Zend Technologies
```

sudo apachectl -M:

```
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 127.0.1.1. Set the 'ServerName' directive globally to suppress this message
Loaded Modules:
 core_module (static)
 so_module (static)
 watchdog_module (static)
 http_module (static)
 log_config_module (static)
 logio_module (static)
 version_module (static)
 unixd_module (static)
 access_compat_module (shared)
 alias_module (shared)
 auth_basic_module (shared)
 authn_core_module (shared)
 authn_file_module (shared)
 authz_core_module (shared)
 authz_host_module (shared)
 authz_user_module (shared)
 autoindex_module (shared)
 deflate_module (shared)
 dir_module (shared)
 env_module (shared)
 filter_module (shared)
 mime_module (shared)
 mpm_prefork_module (shared)
 negotiation_module (shared)
 php7_module (shared)
 reqtimeout_module (shared)
 rewrite_module (shared)
 setenvif_module (shared)
 status_module (shared)
```

Version of glados:
GLaDOS 1.0.4 - 2018

User logged in: admin

[glados_requirements.pdf](https://github.com/imedias/glados/files/2518151/glados_requirements.pdf)
![bildschirmfoto vom 2018-10-26 09-24-04](https://user-images.githubusercontent.com/43335992/47550950-124c8700-d901-11e8-847c-ea3d62bc6ef6.png)

Greetings feschoppe

**Comment by chaoos** [comment]
Thank your for the detailed information!
It looks like a problem with the role based access control system and the binding of the permissions (something went wrong during the installation). Can you attach the output of the following 2 commands?

* `mysql -h localhost -P 3306 -u glados -p -e "use glados; select * from auth_assignment;"`
* `mysql -h localhost -P 3306 -u glados -p -e "use glados; select id,username from user;"`

You have to provide your database password (the one you defined during the installation of glados).
The outputs should look like this:
```
# mysql -h localhost -P 3306 -u glados -p -e "use glados; select * from auth_assignment;"
Enter password:
+-----------+---------+------------+
| item_name | user_id | created_at |
+-----------+---------+------------+
| admin     |       1 | 1540543661 |
| teacher   |       2 | 1540543661 |
+-----------+---------+------------+
```
 and

```
# mysql -h localhost -P 3306 -u glados -p -e "use glados; select id,username from user;"
Enter password:
+----+----------+
| id | username |
+----+----------+
|  1 | admin    |
|  2 | teacher  |
+----+----------+
```

If your output does look like this, please append the output of these 2 commands too:

* `mysql -h localhost -P 3306 -u glados -p -e "use glados; select name,type,description from auth_item"`
* `mysql -h localhost -P 3306 -u glados -p -e "use glados; select * from auth_item_child"`

**Comment by feschoppe** [comment]
I fixed the problem with first purging the mysql-server package and then reinstalling the glados debian package and now it works. It seems to me that it was indeed an issue with the mysql configuration, because the command
`mysql -h localhost -P 3306 -u glados -p -e "use glados; select * from auth_assignment;"`
did not show any output at all.
When I installed the glados package the first time the installation failed. Then I first installed the dependencies separately and then the glados package again and then it worked but apparently the mysql configuration was messed up.

But now it works!
